tree/treeplayer/inc/ROOT/TCutFlowReport.hxx

```diff
@@ -0,0 +1,67 @@
+// Author: Enrico Guiraud, Danilo Piparo CERN  02/2018
+
+/*************************************************************************
+ * Copyright (C) 1995-2016, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TCUTFLOWREPORT
+#define ROOT_TCUTFLOWREPORT
+
+#include "RtypesCore.h"
+#include "TString.h"
+
+#include <string>
+#include <vector>
+
+namespace ROOT {
+
+namespace Detail {
+namespace TDF {
+class TFilterBase;
+} // End NS TDF
+} // End NS Detail
+
+namespace Experimental {
+namespace TDF {
+
+class TCutFlowReport;
```

Comment:
no need for this, friend declarations already forward-declare objects in the same namespace